ServerCore/Pages/Components/PresenceComponent.razor.cs

```diff
@@ -106,7 +109,11 @@ protected override async Task OnParametersSetAsync()
 
             await UpdateModelAsync(teamPuzzleStore.PresentPages);
 
-            await MessageHub.BroadcastPresenceMessageAsync(new PresenceMessage { PageInstance = pageInstance, PuzzleUserId = PuzzleUserId, TeamId = TeamId, PuzzleId = PuzzleId, PresenceType = PresenceType.Active });
+            if (!this.IsReadOnly)
```

Comment:
This check is also needed in DisposeAsync where it shows the player disconnecting from the puzzle